src/averageAges.js

```diff
@@ -54,8 +77,24 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  const filteredPeople = onlyWithSon
+    ? people
+      .filter(person => person.sex === 'm' && people
+        .find(mother => mother.name === person.mother))
+    : people
+      .filter(person => people
+        .find(mother => mother.name === person.mother));
+
+  const diffArr = filteredPeople.map(person =>
+    person.born - people
+      .find(mother => mother.name === person.mother).born);
+
+  function getAverageAge(input) {
+    return input.reduce((sum, age) => sum + age, 0) / input.length;
```

Comment:
You can use `getAverageAge` for other functions, where you have similar calculation